content/blog/intro-to-step-functions/index.md

```diff
@@ -0,0 +1,178 @@
+)
+```
+
+The state machine controls the workflow. It executes using the IAM role we defined and has all the permissions necessary to execute the tasks defined. The `definition` specifies the start state and all the other states. In our case, there’s only one state, `HelloWorld`.
+
+```python
+state_defn = state_machine = sfn.StateMachine('stateMachine',
+   role_arn=iam.sfn_role.arn,
+   definition=hello_world_fn.arn.apply(lambda arn: """{
+       "Comment": "A Hello World example of the Amazon States Language using an AWS Lambda Function",
+       "StartAt": "HelloWorld",
+       "States": {
+           "HelloWorld": {
+               "Type": "Task",
+               "Resource": "%s",
+               "End": true
+           }
+       }
+   }""" % arn)
+)
+
+```
+
+To run the example run and select `yes`.
+
+```bash
+$ pulumi up
+Previewing update (dev):
+
+     Type                     Name                  Plan
+     pulumi:pulumi:Stack      stepfunctions-py-dev
+ +   ├─ aws:iam:Role          lambdaRole            create
+ +   ├─ aws:iam:Role          sfnRole               create
+ +   ├─ aws:iam:RolePolicy    lambdaRolePolicy      create
+ +   ├─ aws:iam:RolePolicy    sfnRolePolicy         create
+ +   ├─ aws:lambda:Function   helloWorldFunction    create
+ +   └─ aws:sfn:StateMachine  stateMachine          create
+
+Outputs:
+  + state_machine_arn: output<string>
+
+Resources:
+    + 6 to create
+    1 unchanged
+
+Do you want to perform this update?
+  yes
+> no
+  details
+```
+
+Select `yes` to run the example, but note that the function doesn’t specify where to return `Hello world!`. We can execute the function from the AWS cli,  but to see how it works, we’ll need to launch the  [AWS Conslole](https://console.aws.amazon.com/states/home#/statemachines/).
+
+Call the function first.
+
+```bash
+aws stepfunctions start-execution --state-machine-arn $(pulumi stack output state_machine_arn)
+```
+
+Then check the AWS console to see if the lambda was executed.
+
+![AWS Severless Console](sf-console-1.png)
+
+Clicking on the `Name` displays the step details, which include the input and out, i.e., ‘Hello world!’
+
+![AWS Severless Console  with 1 step function](sf-console-2.png)
+
+The AWS console provides a graph of your workflow and the details of each step. Clicking on `Output` shows “Hello world” meaning that that the call to the lambda function was successful.
+
+## Add another step
+
+Now that we have a basic flow, let’s add another step. To keep it simple we’ll create another simple function that returns “Nice to meet you.” To do this, make a file called `nice.py` and add the function.
+
+```bash
+def handler(event, context):
+   return "Nice to meet you"
+
+```
+
+We will also need to import `Output` from pulumi along with the other imports. We’ll use it later.
+
+```python
+from pulumi import Output
+```
+
+Next, add the new lambda function. Note that we can reuse the same IAM role and policy we use in the `hello_world_fn` function.
```

Comment:
Nit: Does it make sense to mention the name? e.g. "... new lambda function, `nice_fn`. Note that we can ..."

Reply:
added